config/cache.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * @file
+ * Cache of config variables
+ *
+ * @authors
+ * Copyright (C) 2023 Richard Russon <rich@flatcap.org>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MUTT_CONFIG_CACHE_H
+#define MUTT_CONFIG_CACHE_H
+
+extern const struct Slist *CachedAssumedCharset;
+extern const char *CachedCharset;
```

Comment:
How about we expose functions instead of global variables? It'd give us more freedom to implement them one way or the other, e.g., we wouldn't need a setup function: we could init the backing global on demand.